src/Controls/tests/Core.UnitTests/VisualElementTests.cs

```diff
@@ -175,5 +175,23 @@ public void RectangleGeometrySubscribed()
 			geometry.Rect = new Rect(1, 2, 3, 4);
 			Assert.True(fired, "PropertyChanged did not fire!");
 		}
+
+		[Fact]
+		public async Task ShadowDoesNotLeak()
+		{
+			var shadow = new Shadow
+			{
+				Brush = new SolidColorBrush(Colors.Black),
+				Radius = 12
+			};
+
+			var reference = new WeakReference(new VisualElement { Shadow = shadow });
+
+			await Task.Yield();
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+
+			Assert.False(reference.IsAlive, "VisualElement should not be alive!");
+		}
```

Comment:
We should add this test:

```csharp
[Fact]
public async Task ShadowSubscribed()
{
    var shadow = new Shadow();
	var visual = new VisualElement { Shadow = shadow };

	bool fired = false;
	visual.PropertyChanged += (sender, e) =>
	{
		if (e.PropertyName == nameof(VisualElement.Shadow))
			fired = true;
	};

	await Task.Yield();
	GC.Collect();
	GC.WaitForPendingFinalizers();

	shadow.Radius = 42;
	Assert.True(fired, "PropertyChanged did not fire!");
}
```

I think it will currently fail, until we figure out what to do for the issue seen in #13973.